hoomd/CachedAllocator.h

```diff
@@ -236,18 +216,7 @@ T* CachedAllocator::getTemporaryBuffer(size_t num_elements)
 //                << "reached; removing unused block ("
 //                << float(i->first)/1024.0f/1024.0f << " MB)" << std::endl;
 
-            // transform the pointer to cuda::pointer before calling cuda::free
-            #ifdef __HIP_PLATFORM_HCC__
-            if (m_managed)
-                {
-                // workaround, HIP doesn't fully support managed memory API
-                hipHostFree((void *) i->second);
-                }
-            else
-            #endif
-                {
-                hipFree((void *) i->second);
-                }
```

Comment:
There are a few places where a change like this is made. Can you explain why this change is necessary?

Reply:
in HIP on AMD platforms, managed memory is being emulated as zero copy memory and in the early days of HIP (around version 2.9), actually required one to use the `hipHostFree` API to free managed memory instead of `hipFree`. Now at least there is API parity with CUDA on that aspect.